scripts/normalize-by-median.py

```diff
@@ -43,33 +47,31 @@ def batchwise(coll, size):
 # Returns true if the pair of records are properly pairs
 
 
-# pylint: disable=too-many-locals,too-many-branches
-def normalize_by_median(input_filename, outfp, htable, paired, cutoff,
-                        report_fp=None):
+def WithDiagnostics(ifile, batch_size, fp, paired):
+    """
+    Generator/context manager to do boilerplate output of statistics while
+    normalizing data. Also checks for properly paired data.
+    """
 
-    desired_coverage = cutoff
-    ksize = htable.ksize()
+    index = 0
+    global total, discarded
```

Comment:
I dislike these globals.  Globals in general are bad.  Among other things they make the code non-reusable elsewhere.

Could we make WithDiagnostics an object that had 'total' and 'discarded' attributes, instead? Happy to sketch that out in more detail if you like.